lib/schema_plus/core/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -35,6 +35,24 @@ def drop_table(table_name, options={})
             end
           end
 
+          # replaced version to handle array of columns with expressions
+          def quoted_columns_for_index(column_names, **options)
+            output_columns = []
+
+            column_names.each do |column|
```

Comment:
Is this clearer?

```suggestion
            column_names.flat_map do |column|
              if column.is_a?(String) && column.include?('(')
                column
              else
                quoted = {column.to_sym => quote_column_name(column).dup}
                add_options_for_index_columns(quoted, options).values
              end
            end
```

Reply:
Also, the spec says this is for Postgres, but AFAICS this would apply to all adapters, is that right?

Reply:
I'd agree this would be clearer, but line 44 would need to be `[column]`, no?

Reply:
@lowjoel so, yes while the code is generic in rails (in ConnectionAdaper::SchemaStatements), neither mysql, nor sqlite support expression indexes thus the test can only test the behavior on Postgres.